eq-author-api/src/businessLogic/onPageDeleted.js

```diff
@@ -0,0 +1,23 @@
+const { getPages } = require("../../schema/resolvers/utils");
+
+const onPageDeleted = (ctx, removedPageId) => {
+  const allPages = getPages(ctx);
+
+  allPages.forEach(page => {
+    if (!page.routing) {
+      return;
+    }
+
+    const validRules = page.routing.rules.filter(
+      rule => rule.destination && rule.destination.pageId !== removedPageId
+    );
+
+    if (validRules.length) {
+      page.routing.rules = validRules;
+    } else {
+      page.routing = null;
```

Comment:
Can we not set it to an empty array of rules?

Reply:
No, it needs to null page.routing otherwise we're just left with an else statement but no rules. Setting page.routing to null puts it back into the same state as it was before rules were added.